service/matching/taskQueueManager.go

```diff
@@ -285,13 +311,59 @@ func (c *taskQueueManagerImpl) Start() {
 		return
 	}
 	c.liveness.Start()
-	c.taskWriter.Start()
-	c.taskReader.Start()
+	c.StartDbTaskManagement()
+	c.dbTaskManagementUpdatedAt = time.Now()
 	go c.fetchMetadataFromRootPartitionOnInit(context.TODO())
 	c.logger.Info("", tag.LifeCycleStarted)
 	c.metricScope.IncCounter(metrics.TaskQueueStartedCounter)
 }
 
+func (c *taskQueueManagerImpl) StartDbTaskManagement() {
+	c.enableDbTaskManager = c.config.EnableDbTaskManager()
+	if c.enableDbTaskManager {
+		c.dbTaskManager.Start()
+	} else {
+		// TODO(deprecate)
+		c.taskWriter.Start()
+		c.taskReader.Start()
+		c.initializedError.Get(context.TODO())
+	}
+}
+
+func (c *taskQueueManagerImpl) SwitchDbTaskManagementIfNecessary() bool {
```

Comment:
@yiminc Does switching between old and new implementation has to be done on the fly? Looks like this adds a lot of complexity. Requiring a restart for switching implementation sounds reasonable to me and upon switch task queue lease needs to be renewed anyway I think.

nit: the function should probably be renamed to `isDbTaskManagementEnabled()`

Reply:
No. It is totally fine to require a restart to make the change.